src/smart-components/group/groups.js

```diff
@@ -1,49 +1,65 @@
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
-import PropTypes from 'prop-types';
-import { Link, Route, Switch } from 'react-router-dom';
+import { Link, Route, Switch, useHistory } from 'react-router-dom';
 import { expandable } from '@patternfly/react-table';
 import { Button, Stack, StackItem } from '@patternfly/react-core';
 import AddGroupWizard from './add-group/add-group-wizard';
 import EditGroup from './edit-group-modal';
 import RemoveGroup from './remove-group-modal';
+import { shallowEqual, useSelector, useDispatch } from 'react-redux';
 import { TableToolbarView } from '../../presentational-components/shared/table-toolbar-view';
 import { createRows } from './group-table-helpers';
 import { fetchGroups } from '../../redux/actions/group-actions';
 import Group from './group';
 import { TopToolbar, TopToolbarTitle } from '../../presentational-components/shared/top-toolbar';
-import { defaultSettings } from '../../helpers/shared/pagination';
 import { Section } from '@redhat-cloud-services/frontend-components';
 import Role from '../role/role';
 
 const columns = [{ title: 'Name', cellFormatters: [ expandable ]}, 'Description', 'Members', 'Last modified' ];
 
-const Groups = ({ fetchGroups, isLoading, pagination, history: { push }, groups, userIdentity }) => {
+const Groups = () => {
   const [ filterValue, setFilterValue ] = useState('');
   const [ selectedRows, setSelectedRows ] = useState([]);
   const [ removeGroupsList, setRemoveGroupsList ] = useState([]);
 
+  const dispatch = useDispatch();
+  const { groups, pagination, userIdentity, isLoading } = useSelector(({ groupReducer: { groups, isLoading }}) => ({
+    groups: groups.data,
+    pagination: groups.meta,
+    userIdentity: groups.identity,
+    isLoading
+  }), shallowEqual);
+
   useEffect(() => {
-    fetchGroups({ ...pagination, name: filterValue });
+    dispatch(fetchGroups({ ...pagination, name: filterValue }));
   }, []);
 
-  const fetchData = (config) => {
-    fetchGroups(config);
-  };
+  const history = useHistory();
 
   const setCheckedItems = (newSelection) => {
     setSelectedRows((rows) => newSelection(rows).map(({ uuid, name }) => ({ uuid, label: name })));
   };
 
   const routes = () => <Fragment>
-    <Route exact path="/groups/add-group" render={ props => <AddGroupWizard { ...props } postMethod={ fetchData } /> } />
-    <Route exact path="/groups/edit/:id" render={ props => <EditGroup { ...props } postMethod={ fetchData } isOpen/> } />
-    <Route exact path="/groups/removegroups" render={ props => <RemoveGroup { ...props } postMethod={ (ids) => {
-      fetchData();
-      setSelectedRows(selectedRows.filter(row => (!ids.includes(row.uuid))));
-      setFilterValue('');
-    } } isModalOpen groupsUuid={ removeGroupsList } /> } />
+    <Route exact path="/groups/add-group"
+      render={ props => <AddGroupWizard { ...props } postMethod={ config => dispatch(fetchGroups(config)) } /> }
```

Comment:
While you are doing this, you should use the children syntax
```jsx
<Route>
  <AddGroupWizard postMethod={ config => dispatch(fetchGroups(config)) } />
</Route>
```
The `{...props}` should be replaced inside the component with hooks like `useHistory`, `useRouteMatch`, etc.